diff --git a/node_modules/@rspress/theme-default/dist/index.d.ts b/node_modules/@rspress/theme-default/dist/index.d.ts
index 770226b..f8c4be3 100644
--- a/node_modules/@rspress/theme-default/dist/index.d.ts
+++ b/node_modules/@rspress/theme-default/dist/index.d.ts
@@ -566,4 +566,4 @@ export declare function useSidebarData(): SidebarData;
  */
 export declare const useThemeState: () => readonly [ThemeValue, (value: ThemeValue, storeValue?: ThemeConfigValue) => void];
 
-export { }
+export declare function renderInlineMarkdown(text: string): { children: string | null } | { dangerouslySetInnerHTML: { __html: string } }
