diff --git a/node_modules/eslint-plugin-mdx/lib/rules/remark.js b/node_modules/eslint-plugin-mdx/lib/rules/remark.js
index b3caaca..c1a88a3 100644
--- a/node_modules/eslint-plugin-mdx/lib/rules/remark.js
+++ b/node_modules/eslint-plugin-mdx/lib/rules/remark.js
@@ -11,9 +11,9 @@ export const remark = {
         fixable: 'code',
     },
     create(context) {
-        const filename = context.getFilename();
+        const filename = context.filename ?? context.getFilename();
         const extname = path.extname(filename);
-        const sourceCode = context.getSourceCode();
+        const sourceCode = context.sourceCode ?? context.getSourceCode();
         const { extensions, markdownExtensions, ignoreRemarkConfig, remarkConfigPath, } = {
             ...context.parserOptions,
             ...context.languageOptions?.parserOptions,
@@ -32,7 +32,7 @@ export const remark = {
                 const { messages, content: fixedText } = performSyncWork({
                     filePath: getPhysicalFilename(filename),
                     code: sourceText,
-                    cwd: context.getCwd(),
+                    cwd: context.cwd ?? context.getCwd(),
                     isMdx,
                     process: true,
                     ignoreRemarkConfig,
